db/sqlanalyze.c

```diff
@@ -759,20 +761,51 @@ static int analyze_rename_table_for_backup_stats(char *table, struct sqlclntstat
 
 int analyze_regular_table(table_descriptor_t *td, struct sqlclntstate *clnt, char *zErrTab, size_t nErrTab) {
     int rc = 0;
+    int have_schema_lk = 0;
     int sampled_table = 0;
-    struct dbtable *table = get_dbtable_by_name(td->table);
-    rc = analyze_rename_table_for_backup_stats(td->table, clnt, zErrTab);
+    tran_type * trans = NULL;
 
-    if (rc) 
+    rc = analyze_rename_table_for_backup_stats(td->table, clnt, zErrTab);
```

Comment:
I initially tried to call this function in the table-locked section, but this causes a deadlock because this function blocks on threads that try to acquire write locks.